### PR TITLE
DAOS-10456 control: Enable VMD by default

### PR DESCRIPTION
**Pull request by tanabarr**
Currently VMD has to be manually enabled by setting enable_vmd: true in
server config file, the downside is that on VMD enabled systems storage
scan will fail if not manually enabled. This also relates to the use case
of DAOS RPM installation on a VMD-enabled system, where discovery mode
will fail with a blank server config file (default on first install)
until the user manually creates a server config file with enable_vmd: true
in it. Enable by default to avoid this issue.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/tests/ftest/launch.py
Comment (phender) [inline src/tests/ftest/launch.py:758]: This code is required for CI.  When we run on non-VMD enabled clusters we need launch.py to correctly set the enable_vmd/disable_vmd variable.

File: src/tests/ftest/util/server_utils_params.py
Comment (phender) [inline src/tests/ftest/util/server_utils_params.py:122]: This code is also required for CI.  This class provides the ability for the test code to manipulate any server config file parameter so at the very least it should include `enable_vmd`/`disable_vmd` to allow us to write tests to verify the parameter.

**Review by phender — CHANGES_REQUESTED** [review]
Personally I find `enable_vmd: true` to be much clearer than `disable_vmd: false`.  Why don't we just change the default value of `enable_vmd: false` to `enable_vmd: true`?  I'm also not sure how much this change is really needed for functional tests as launch.py and server_utils_params.py already handle setting the default correctly while still providing the possibility for a test to manipulate the setting if needed.

File: src/tests/ftest/launch.py
Comment (tanabarr) [reply src/tests/ftest/launch.py:758]: can you explain why that is still needed, with this change daos_server shouldn't need-to-know whether the bdev_list PCI addresses are VMD endpoints or not, the only time we should need to set disable_vmd is in the case where VMD is enabled on the system but we want to use non-VMD enabled SSDs. as far as I know that shouldn't happen in CI.

File: src/tests/ftest/util/server_utils_params.py
Comment (phender) [reply src/tests/ftest/util/server_utils_params.py:122]: So after getting clarification I would just update this comment to provide the ability to set the value, but not include it in the server config by default with:
```
self.disable_vmd = BasicParameter(None)
```

File: src/tests/ftest/launch.py
Comment (phender) [reply src/tests/ftest/launch.py:758]: Ok, then I recant my objection.  Thanks for the clarification.

**Comment by phender** [comment]
> > Personally I find `enable_vmd: true` to be much clearer than `disable_vmd: false`. Why don't we just change the default value of `enable_vmd: false` to `enable_vmd: true`?
> 
> Without commenting on the rest of the changes, I'll just say here that the intent behind this change is about indicating what the default/normal mode of operation is. Using `enable_vmd` indicates that VMD is an additional/optional feature that isn't normally used. Whereas `disable_vmd` indicates that VMD is by default enabled as part of the normal mode of operation, and the config option is there to disable it if it's causing problems or the admin knows that they really don't want it to be enabled for some reason.

That make sense, thanks.

**Review by kjacque — COMMENTED** [review]
Looks good from my side. I'll approve once Phil's issue is resolved.

**Review by phender — APPROVED** [review]

**Comment by tanabarr** [comment]
waiting for CI provisioning issues to be resolved before I rerun

**Review by kjacque — APPROVED** [review]

**Pull request merged**
